model/openapi.yml

```diff
@@ -139,6 +139,14 @@ components:
         - type
         - metadata
       properties:
+        id:
+          type: string
+          description: |
+            The ID of the agent.
+            This is the ID that will be used to reference this agent, if no ID is passed one will be generated.
+            If another agent is enrolled with the same ID the other agent will no longer be able to communicate,
+            this new agent is considered a replacement of the other agent. The other agent will be able to continue
```

Comment:
> The other agent will be able to continue sending data to ES.

How do the ES API keys eventually get revoked? This normally happens through unenrolling an agent via the UI, but with replacement of the agent, you lose that.

Should replacing an agent revoke it's ES API keys to treat it like a force unenroll but triggered through fleet server?

Reply:
I think I maybe misunderstood what happens the first time, the ES API keys are tied to the agent ID so they are kept alive because they are reused.

So this is documenting that this feature only works as expected if the agent you are replacing stops running. Otherwise you will have two agents ingesting data but one of them is unmanaged and has had its Fleet API key revoked. Correct?

Reply:
That is correct. It will be able to keep ingesting data, which we want for a short period, but will not be able to communicate with Fleet. Then its up to the deployer to turn it off when the other one is working.

This works for Kubernetes, as it will allow a new pod to spawn while the previous version is still running. Once the new version is up and running and healthy then the other pod will be stopped.